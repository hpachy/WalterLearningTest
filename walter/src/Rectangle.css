.App > .rectangle {
    display:inline-block;
    width: 60px;
    font-size: 2em;
    flex: 1 1 calc(100% / 6 - 0.4em);
    outline: 0.08em solid silver;
    margin: 0.2em;
    display: flex;
    cursor: default;
  }
  
  .App > .rectangle.hidden {
    background: silver;
  }

  .App > .rectangle.red {
    outline-color: red;
    background: red;
  }
  .App > .rectangle.orange {
    outline-color: orange;
    background: orange;
  }
  .App > .rectangle.blue {
    outline-color: blue;
    background: blue;
  }
  
  .App > .rectangle.visible {
    cursor: not-allowed;
  }
  
  .App > .Rectangle > .compter {
    margin: auto;
  }