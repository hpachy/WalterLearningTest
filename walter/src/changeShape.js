export default function changeShape(Timer) {
    if ( Timer > 20 )
      return 'blue'
    if ( Timer > 12 && Timer <= 20 )
      return 'orange'
    if ( Timer >= 0 && Timer <= 12 )
      return 'red'
    return 'hidden'
}