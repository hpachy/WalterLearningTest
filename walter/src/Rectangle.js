import React from 'react'
import PropTypes from 'prop-types';

import './Rectangle.css'


const Rectangle = ({compter, currentTime, feedback}) => (
    <div className={`rectangle ${feedback}`}>
        <span className="compter">{('0' + compter).slice(-2)}</span>
    </div>
)

Rectangle.prototype = {
    card: PropTypes.string.isRequired,
    feedback: PropTypes.oneOf([
        'visible',
        'justMatched',
        'justMismatched' 
    ]).isRequired, 
}

export default Rectangle