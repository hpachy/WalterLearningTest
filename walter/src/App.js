import React, { Component } from 'react'
import './App.css';
import Rectangle from './Rectangle'
import changeShape from './changeShape'

class App extends Component {
  state = {
    Timer: 40,
    currentTime: new Date().getSeconds(),
    interval: null
  }
  componentDidMount() {
   var interval = setInterval(() => this.timeLaucher(this.state.Timer), 1000)
   this.setState({interval : interval})
  }

  timeLaucher = time => {
    const {Timer, currentTime} = this.state
    var tmp = new Date().getSeconds()
    if (Timer === 0){
      this.setState({Timer: 0, currentTime: NaN})
      clearInterval(this.state.interval)
      return
    }
    if (currentTime !== tmp){
      this.setState({Timer: Timer - 1, currentTime: tmp})
    }
    return tmp
  }

  render() {
    const {Timer} = this.state
    return (
    <div className="App">
      <Rectangle 
        compter={Timer}
        feedback={changeShape(Timer)}
      />
    </div>
  );
}
}

export default App;
