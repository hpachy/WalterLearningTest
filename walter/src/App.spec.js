import React from 'react'
import {shallow} from 'enzyme'
import App, {SYMBOLS} from './App'
import { expect } from 'chai'
import Rectangle from './Rectangle'
import sinon from 'sinon'
import changeShape from './changeShape'

Rectangle
describe('<App />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)

    expect(wrapper.find('Rectangle'))
  })

  it('contain Rectangle is blue', () => {
    const wrapper = shallow(<Rectangle compter={40} feedback={changeShape(40)} />).getElement()

    expect(wrapper).to.matchSnapshot()
  })

  it('contain Rectangle is orange', () => {
    const wrapper = shallow(<Rectangle compter={20} feedback={changeShape(20)} />).getElement()

    expect(wrapper).to.matchSnapshot()
  })

  it('contain Rectangle is red', () => {
        const wrapper = shallow(<Rectangle compter={12} feedback={changeShape(12)} />).getElement()
        expect(wrapper).to.matchSnapshot()
  })

  it('under 10 we still have 2 numbers', () => {
    const wrapper = shallow(<Rectangle compter={8} feedback={changeShape(8)} />).getElement()
    expect(wrapper).to.matchSnapshot()
})
})
