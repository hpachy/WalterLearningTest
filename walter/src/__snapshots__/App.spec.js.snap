// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> contain Rectangle is blue 1`] = `
<div
  className="rectangle blue"
>
  <span
    className="compter"
  >
    40
  </span>
</div>
`;

exports[`<App /> contain Rectangle is orange 1`] = `
<div
  className="rectangle orange"
>
  <span
    className="compter"
  >
    20
  </span>
</div>
`;

exports[`<App /> contain Rectangle is red 1`] = `
<div
  className="rectangle red"
>
  <span
    className="compter"
  >
    12
  </span>
</div>
`;

exports[`<App /> under 10 we still have 2 numbers 1`] = `
<div
  className="rectangle red"
>
  <span
    className="compter"
  >
    08
  </span>
</div>
`;
